desc/objectives/objective_funs.py

```diff
@@ -387,7 +390,7 @@ def x(self, *things):
         """Return the full state vector from the Optimizable objects things."""
         # TODO: also check resolution etc?
         things = things or self.things
-        assert all([type(t1) == type(t2) for t1, t2 in zip(things, self.things)])
+        assert all([type(t1) is type(t2) for t1, t2 in zip(things, self.things)])
```

Comment:
is this to allow subclasses to match against their parent class types?

Reply:
This was basically just to shut up my linter. Using `is` is the more pythonic/correct way to check types like this, since types are singleton objects so comparing identity is the same as equality